src/map.cpp

```diff
@@ -2726,22 +2726,59 @@ void map::drop_items( const tripoint &p )
     // rather than disappearing if it would be overloaded
 
     tripoint below( p );
+    int height_fallen = 0;
     while( !has_floor( below ) ) {
         below.z--;
+        height_fallen++;
     }
 
     if( below == p ) {
         return;
     }
 
-    for( const item &i : items ) {
-        // TODO: Bash the item up before adding it
-        // TODO: Bash the creature, terrain, furniture and vehicles on the tile
+    float damage = 0.0f;
+    for( item &i : items ) {
+        units::mass wt_dropped = i.weight();
+        damage += 10 * to_kilogram( wt_dropped ) * height_fallen;
+
         add_item_or_charges( below, i );
     }
 
-    // Just to make a sound for now
-    bash( below, 1 );
+    // Bash creature standing below
+    Creature *creature_below = get_creature_tracker().creature_at( below );
+    if( creature_below ) {
+        float dodge_mod = creature_below->dodge_roll();
+        dodge_mod = dodge_mod == 0 ? 1 : dodge_mod / 2;
+
+        int creature_hit_chance = rng( 0, 100 );
+        creature_hit_chance *= dodge_mod;
+
+        if( creature_hit_chance < 15 ) {
+            creature_below->deal_damage( nullptr, bodypart_id( "head" ), damage_instance( damage_bash,
+                                         damage ) );
+        } else if( creature_hit_chance < 30 ) {
+            creature_below->deal_damage( nullptr, bodypart_id( "torso" ), damage_instance( damage_bash,
+                                         damage ) );
+        } else if( creature_hit_chance < 65 ) {
+            creature_below->deal_damage( nullptr, bodypart_id( "arm_l" ), damage_instance( damage_bash,
+                                         damage ) );
+        } else if( creature_hit_chance < 100 ) {
```

Comment:
100% chance of hitting a target that isn't dodging isn't a reasonable starting point.
I think starting with,  "you can make the item land at a random position in the tile at throwing 0" is a good starting point,  then the hit chance is proportion of tile occupied.
Higher throwing skill would increase this chance,  higher target dodge would reduce it.

Reply:
This is swinging too far in the other direction, a dodge_roll of 2 cuts the change to hit the target in half and it gets worse from there. Note, a monster "dodge" stat of 1 yields a dodge_roll() of 5, so it goes from "can't dodge at all" for about a 50% hit chance for most monsters to a 1/10 hit change for the tiniest amount of dodging ability, and likely several times that.

Reply:
Fixed. According to my calculations:
t,d                 % hit chance
0,0                60
0,1                40
0,2                36
1,0                76
2,0                93
...